### PR TITLE
[L1T] add CPPF DQM module

### PR DESCRIPTION
**Pull request by cobby319**
#### PR description:

[L1T] implement CPPF Input and Output Occupancy DQM 

#### PR validation:

Test using standard Release Validation workflow: runTheMatrix.py --what standard -l 136.867
see full introduction here: https://twiki.cern.ch/twiki/bin/view/Sandbox/L1TOccupancyPlots

### CODE REVIEW

**Comment by jfernan2** [comment]
@cobby319 we don't have you identified as a DQM developer, and git does not tell us your name. Can you please add yourself with your git name in the corresponding e-group? https://twiki.cern.ch/twiki/bin/viewauth/CMS/DQMContacts

**Comment by jfernan2** [comment]
@cobby319 Some PR description and setting clearly the subsystem name in the Title is desired too

**Comment by cobby319** [comment]
> @cobby319 Some PR description and setting clearly the subsystem name in the Title is desired too

Hi, Here is my information in egroup
Hanwen Wang | allenwang@buaa.edu.cn | hanwen

**Comment by jfernan2** [comment]
Thanks @cobby319  but can you please add to the title [L1T] to clearly identify the subsystem beyond the subsystem configuration key CPPF?

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@cobby219 this PR is adding 110 histograms to dedicated workflows, can you please check the content is OK across the desired workflows? Some of workflows have plots with zero entries
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_0_X_2019-09-24-1100+894d24/33583/dqm-histo-comparison-summary.html

**Comment by cobby319** [comment]
> @cobby219 this PR is adding 110 histograms to dedicated workflows, can you please check the content is OK across the desired workflows? Some of workflows have plots with zero entries
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_0_X_2019-09-24-1100+894d24/33583/dqm-histo-comparison-summary.html

Sorry I don't know what these baseline numbers represent for. But for CPPF occupancy histos, we only expect entries from real Data. For MC, there are certainly no entries.

**Comment by cobby319** [comment]
> @cobby219 this PR is adding 110 histograms to dedicated workflows, can you please check the content is OK across the desired workflows? Some of workflows have plots with zero entries
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_0_X_2019-09-24-1100+894d24/33583/dqm-histo-comparison-summary.html

Ok, I've compared with the numbers and their corresponding datasets at here https://github.com/cms-sw/cmssw/blob/32a2333cc733889305d4ff6396da6315d31fc22e/Configuration/PyReleaseValidation/python/relval_standard.py.

It seems that most zero-entry histos are from MC. 
CPPF occupancy plots are filled when you get Muons in real Data events.

**Comment by jfernan2** [comment]
@cobby319  in this case you should disable your module for MC workflows to not pollute DQM with empty plots

File: DQM/L1TMonitor/src/L1TStage2CPPF.cc [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+
+L1TStage2CPPF::L1TStage2CPPF(const ParameterSet& ps)
+    : rpcdigiSource_(ps.getParameter<InputTag>("rpcdigiSource")),
+      rpcdigiSource_token_(consumes<RPCDigiCollection>(ps.getParameter<InputTag>("rpcdigiSource"))),
+      cppfdigiSource_token_(consumes<l1t::CPPFDigiCollection>(ps.getParameter<InputTag>("rpcdigiSource"))) {
+  // verbosity switch
+  verbose_ = ps.getUntrackedParameter<bool>("verbose", false);
+
+  if (verbose_)
+    cout << "L1TStage2CPPF: constructor...." << endl;
```
Comment (fabiocos) [inline DQM/L1TMonitor/src/L1TStage2CPPF.cc]: @cobby319 please do not use ```cout``` in module that should run in production, as it is not thread-safe. Use the ```MessageLogger``` instead

File: DQM/L1TMonitor/src/L1TStage2CPPF.cc [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+      CPPFOutputNormOccupDisk[d + offset] = ibooker.book2D(
+          histoName.str().c_str(), histoName.str().c_str(), 36, 0.5, 36.5, numberOfRings_, 0.5, numberOfRings_ + 0.5);
+
+      rpcUtils.labelXAxisSegment(CPPFOutputNormOccupDisk[d + offset]);
+      rpcUtils.labelYAxisRing(CPPFOutputNormOccupDisk[d + offset], numberOfRings_, useRollInfo_);
+    }
+
+    for (int ring = 2; ring <= 3; ring++) {
+      for (int region = -1; region < 2; region += 2) {
+        /*
```
Comment (fabiocos) [inline DQM/L1TMonitor/src/L1TStage2CPPF.cc]: @cobby319 why this commented piece?

File: DQM/L1TMonitor/src/L1TStage2CPPF.cc [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+    //yBin = (detId.ring()-1)*3-3+1;  // no roll information
+    yBin = detId.ring() - 1;
+    int ring = detId.ring();
+
+    wheelOrDiskType = "Disk";
+    wheelOrDiskNumber = region * (int)detId.station();
+
+    if (region != 0) {
+      /*
+      os.str("");
```
Comment (fabiocos) [inline DQM/L1TMonitor/src/L1TStage2CPPF.cc]: here as well

**Comment by fabiocos** [comment]
@cobby319 I had a few trivial comments that are still pending, what are the plans for this PR? Is there still interest in it?

**Comment by cobby319** [comment]
> @cobby319 I had a few trivial comments that are still pending, what are the plans for this PR? Is there still interest in it?

Sorry for the late reply.
Actually we just discussed about this PR.
Now we are focusing on making this CPPF DQM module work for MC sample as well.
It's not a easy task and it may take some time.

**Comment by fabiocos** [comment]
@cobby319 any news about this PR?

**Comment by fabiocos** [comment]
hold

due to lack of activity

**Comment by silviodonato** [comment]
@cobby319  , unless you will update this PR in the next weeks, I would suggest to close this PR and open a new PR as soon as you have news.

**Comment by silviodonato** [comment]
Please let me know as soon as you need to reopen this PR.

**Pull request closed without merging**